cdap-app-fabric/src/main/java/io/cdap/cdap/app/preview/DefaultPreviewRunnerManager.java

```diff
@@ -0,0 +1,216 @@
+import io.cdap.cdap.internal.provision.ProvisionerModule;
+import io.cdap.cdap.logging.guice.LocalLogAppenderModule;
+import io.cdap.cdap.messaging.guice.MessagingServerRuntimeModule;
+import io.cdap.cdap.metadata.MetadataReaderWriterModules;
+import io.cdap.cdap.metrics.guice.MetricsClientRuntimeModule;
+import io.cdap.cdap.security.auth.context.AuthenticationContextModules;
+import io.cdap.cdap.security.guice.preview.PreviewSecureStoreModule;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.tephra.TransactionSystemClient;
+import org.apache.twill.discovery.DiscoveryService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Service for managing {@link PreviewRunnerService}.
+ */
+public class DefaultPreviewRunnerManager extends AbstractIdleService implements PreviewRunnerManager {
+  private static final Logger LOG = LoggerFactory.getLogger(DefaultPreviewRunnerManager.class);
+
+  private final CConfiguration previewCConf;
+  private final Configuration previewHConf;
+  private final SConfiguration previewSConf;
+  private final int maxConcurrentPreviews;
+  private final DiscoveryService discoveryService;
+  private final DatasetFramework datasetFramework;
+  private final SecureStore secureStore;
+  private final TransactionSystemClient transactionSystemClient;
+  private Injector previewInjector;
+  private final PreviewRunnerModule previewRunnerModule;
+  private final Map<String, PreviewRunnerService> previewPollers;
+  private final LevelDBTableService previewLevelDBTableService;
+  private final PreviewRequestFetcherFactory previewRequestFetcherFactory;
+
+  @Inject
+  DefaultPreviewRunnerManager(
+    @Named(PreviewConfigModule.PREVIEW_CCONF) CConfiguration previewCConf,
+    @Named(PreviewConfigModule.PREVIEW_HCONF) Configuration previewHConf,
+    @Named(PreviewConfigModule.PREVIEW_SCONF) SConfiguration previewSConf,
+    SecureStore secureStore, DiscoveryService discoveryService,
+    @Named(DataSetsModules.BASE_DATASET_FRAMEWORK) DatasetFramework datasetFramework,
+    TransactionSystemClient transactionSystemClient,
+    @Named(PreviewConfigModule.PREVIEW_LEVEL_DB) LevelDBTableService previewLevelDBTableService,
+    PreviewRunnerModule previewRunnerModule, PreviewRequestFetcherFactory previewRequestFetcherFactory) {
+    this.previewCConf = previewCConf;
+    this.previewHConf = previewHConf;
+    this.previewSConf = previewSConf;
+    this.datasetFramework = datasetFramework;
+    this.secureStore = secureStore;
+    this.discoveryService = discoveryService;
+    this.transactionSystemClient = transactionSystemClient;
+    this.maxConcurrentPreviews = previewCConf.getInt(Constants.Preview.CACHE_SIZE, 10);
+    this.previewPollers = new ConcurrentHashMap<>();
```

Comment:
Is this class only used in standalone?? Because in distributed mode, we shouldn't have this.

Reply:
For simplicity kept the same class in distributed mode as well. In Distributed mode `maxConcurrentPreviews` will be set to 1. 

We can adjust the guice bindings if required in the PR which will have distributed mode changes.